src/main/java/org/wso2/carbon/inbound/poll/dbeventlistener/DBEventPollingConsumer.java

```diff
@@ -0,0 +1,380 @@
+
+        }
+        return columnValue;
+    }
+
+    /**
+     * Build the SELECT query as string
+     * @param tableName
+     * @param filteringCriteria
+     * @param filteringColumnName
+     * @param lastUpdatedTimestampFromRegistry
+     * @return query
+     */
+    private String buildQuery(String tableName, String filteringCriteria, String filteringColumnName,
+            String lastUpdatedTimestampFromRegistry) {
+        if (filteringCriteria.equals(DBEventConstants.DB_FILTERING_BY_TIMESTAMP)) {
+            return "SELECT * FROM " + tableName + " WHERE " + filteringColumnName + " > '"
+                    + lastUpdatedTimestampFromRegistry + "' ORDER BY " + filteringColumnName;
+        } else if (filteringCriteria.equals(DBEventConstants.DB_FILTERING_BY_BOOLEAN)){
+            return "SELECT * FROM " + tableName + " WHERE " + filteringColumnName + "='true'";
+        } else {
+            return "SELECT * FROM " + tableName;
+        }
+    }
+
+    @Override
+    public void destroy() {
+        try {
+            if (connection != null) {
+                connection.close();
+            }
+        } catch (SQLException sqle) {
+            log.error("Error while destroying the connection to '" + dbURL + "'", sqle);
+        }
+    }
+
+    /**
+     * Create the database connection
+     */
+    private void createConnection() {
+        try {
+            Class.forName(driverClass);
+            connection = DriverManager.getConnection(dbURL, dbUsername, dbPassword);
+        } catch (ClassNotFoundException e) {
+            log.error("Unable to find the driver class " + driverClass, e);
+        } catch (SQLException e) {
+            log.error("Error while creating the connection to " + dbURL, e);
+        }
+    }
+
+    @Override
+    public Object poll() {
+        try {
+            if (connection == null || connection.isClosed()) {
+                createConnection();
+            }
+            fetchDataAndInject();
+        } catch (SQLException e) {
+            log.error("Error while checking the connection.", e);
+        }
+        return null;
+    }
+
+    /**
+     * Check whether the message should be rolled back or not.
+     */
+    private boolean isRollback(MessageContext msgCtx) {
+        Object rollbackProp = msgCtx.getProperty(DBEventConstants.SET_ROLLBACK_ONLY);
+        if (rollbackProp != null) {
+            if ((rollbackProp instanceof Boolean && ((Boolean) rollbackProp))
+                    || (rollbackProp instanceof String && Boolean.valueOf((String) rollbackProp))) {
+                return true;
+            }
+            return false;
+        }
+        return false;
+    }
+
+    /**
+     * @return status of the connection
+     */
+    private boolean isConnectionAlive() {
+        Statement statement = null;
+        ResultSet rs = null;
+        try {
+            statement = connection.prepareStatement(connectionValidationQuery);
```

Comment:
Validation query is an optional parameter. We need to handle if the validation query is empty.